pkg/k8shandler/cluster.go

```diff
@@ -223,6 +227,72 @@ func (cState *ClusterState) removeStaleNodes(dpl *v1alpha1.Elasticsearch) error
 	return nil
 }
 
+func (cState *ClusterState) restartCluster(dpl *v1alpha1.Elasticsearch, owner metav1.OwnerReference) error {
+	node := upgradeInProgress(dpl)
+	var err error
+	if node == nil {
+		// don't attempt to restart the cluster unless cluster health is green
+		if ok := canRestartCluster(dpl); !ok {
+			return fmt.Errorf("Cluster Rolling Restart requested but cluster isn't ready")
+		}
+		node, err = cState.beginUpgrade(dpl, owner)
+		logrus.Infof("rolling upgrade: began upgrading node: %s", node.PodName)
+	}
+	// find a pod that can handle requests
+	// in a single-master deployment the operator will complain about not having any master pods
+	masterPod, err := getRunningMasterPod(dpl.Name, dpl.Namespace)
+	if err != nil {
+		return nil
+	}
+	// wait for node to start and rejoin the cluster
+	if rejoined := nodeRejoinedCluster(dpl, masterPod); !rejoined {
+		logrus.Infof("rolling upgrade: waiting for node '%s' to rejoin the cluster...", node.PodName)
+		return nil
+	}
+	// enable shard allocation
+	if err = enableShardAllocation(dpl, masterPod); err != nil {
```

Comment:
the elasticsearch docs suggest waiting for the cluster to be `yellow` before you reenable shard allocation to indicate that the node that just rejoined has recovered all primary shards it has stored. I think this is so the cluster doesn't try to recreate primary shards that haven't been recovered yet.

Reply:
Can you post link to that part of the es docs? I can't find it. Thank you.

Reply:
I can't seem to find that wording anymore... so I guess we don't need to wait for `yellow` first?